### PR TITLE
[12.0][MIG] account invoice merge

### PR DESCRIPTION
**Pull request by sylvainvh**
Migration of the module account_invoice_merge

### CODE REVIEW

File: account_invoice_merge/__manifest__.py [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+# Copyright 2019 Okia SPRL
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html).
+
+{
+    'name': 'Account Invoice Merge',
+    'version': '12.0.1.0.1',
+    'category': 'Finance',
+    'summary': "Merge invoices in draft",
+    'author': "Elico Corp,Odoo Community Association (OCA)",
+    'website': 'http://www.openerp.net.cn',
```
Comment (nikul-serpentcs) [inline account_invoice_merge/__manifest__.py:15]: @sylvainvh  URL should be `https://github.com/OCA/account-invoicing`

File: account_invoice_merge/tests/test_account_invoice_merge.py [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+        })
+        return partner
+
+    def _create_inv_line(self, account_id):
+        inv_line = self.inv_line_model.create({
+            'name': 'test invoice line',
+            'account_id': account_id,
+            'quantity': 1.0,
+            'price_unit': 3.0,
+            'product_id': self.env.ref('product.product_product_8').id
```
Comment (nikul-serpentcs) [inline account_invoice_merge/tests/test_account_invoice_merge.py:55]: direct use  `self.ref('product.product_product_8')` no need to `.id`

**Review by nikul-serpentcs — COMMENTED** [review]
nitpicking

**Review by bjeficent — APPROVED** [review]
Functional test on runbot, LGTM :+1:

File: account_invoice_merge/tests/test_account_invoice_merge.py (hunk 2 above)
Comment (sylvainvh) [reply account_invoice_merge/tests/test_account_invoice_merge.py:55]: Hum, I'm not fan to pass the record himself to create or edit a record. For me, product_id should be the ID and not record.

File: account_invoice_merge/__manifest__.py [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+# Copyright 2010-2011 Elico Corp.
+# Copyright 2016 Acsone (https://www.acsone.eu/)
+# Copyright 2017 Eficent Business and IT Consulting Services S.L.
+#   (http://www.eficent.com)
+# Copyright 2019 Okia SPRL
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html).
+
+{
+    'name': 'Account Invoice Merge',
+    'version': '12.0.1.0.1',
```
Comment (Saran440) [inline account_invoice_merge/__manifest__.py:11]: version should be start at `12.0.1.0.0`

**Comment by flotho** [comment]
hi @sylvainvh 

any news on this PR ?

regards

File: account_invoice_merge/models/account_invoice.py [hunk 4]
```diff
@@ -0,0 +1,218 @@
...
+                    field_val = ((6, 0, tuple([v.id for v in field_val])),)
+                list_key.append((field, field_val))
+            list_key.sort()
+            return tuple(list_key)
+
+        # compute what the new invoices should contain
+        new_invoices = {}
+        draft_invoices = [invoice
+                          for invoice in self
+                          if invoice.state == 'draft']
```
Comment (thomaspaulb) [inline account_invoice_merge/models/account_invoice.py]: Perhaps you can include [this](https://github.com/OCA/account-invoicing/pull/732) here also.

**Review by anajuaristi — APPROVED** [review]

**Comment by anajuaristi** [comment]
Any problem to merge this? LGTM :+1:

File: account_invoice_merge/models/account_invoice.py (hunk 4 above)
Comment (mourad-ehm) [reply account_invoice_merge/models/account_invoice.py]: I propose to add get draft invoices & _dirty_check overridable in this PR 
https://github.com/sylvainvh/account-invoicing/pull/1/files

**Review by marcelsavegnago — APPROVED** [review]

**Review by mourad-ehm — APPROVED** [review]

**Review by chafique-delli — APPROVED** [review]

**Comment by mourad-ehm** [comment]
Hi @pedrobaeza,
Could you merge this PR thanks

**Comment by pedrobaeza** [comment]
As stated other times, I don't like the merge approach through this module (and the need of extension ones), so I'm not merging a module that I haven't reviewed nor like. Other PSC member may do it.

**Comment by mourad-ehm** [comment]
Who can do it ?

**Comment by pedrobaeza** [comment]
https://odoo-community.org/psc-teams/accounting-9

**Comment by mourad-ehm** [comment]
Thanks

**Comment by mourad-ehm** [comment]
Hi @sbidoul @kittiu , 
Could you merge this PR please ?

**Comment by kittiu** [comment]
Function tested LGTM.

**Comment by mourad-ehm** [comment]
thanks @kittiu

**Pull request merged**